test/unit/layerupdatestrategy.js

```diff
@@ -0,0 +1,134 @@
+import * as THREE from 'three';
+import assert from 'assert';
+import TileMesh from 'Core/TileMesh';
+import Extent from 'Core/Geographic/Extent';
+import OBB from 'Renderer/OBB';
+import Layer from 'Layer/Layer';
+import Source from 'Source/Source';
+import { STRATEGY_DICHOTOMY, STRATEGY_PROGRESSIVE, STRATEGY_GROUP, chooseNextLevelToFetch } from 'Layer/LayerUpdateStrategy';
+import LayerUpdateState from 'Layer/LayerUpdateState';
+import MaterialLayer from 'Renderer/MaterialLayer';
+
+describe('Handling no data source error', function () {
+    // Misc var to initialize a TileMesh instance
+    const geom = new THREE.Geometry();
+    geom.OBB = new OBB(new THREE.Vector3(), new THREE.Vector3(1, 1, 1));
+
+    const extent = new Extent('EPSG:4326', 6.227531433105469, 6.227874755859375, 44.93614196777344, 44.936485290527344);
+    extent.zoom = 19;
+    const material = {};
+
+    // Mock scheduler
+    const context = {
+        view: {
+            notifyChange: () => true,
+        },
+        scheduler: {
+            commands: [],
+            execute: (cmd) => {
+                context.scheduler.commands.push(cmd);
+                return new Promise(() => { /* no-op */ });
+            },
+        },
+    };
+
+    const source = new Source({
+        url: 'http://',
+        projection: 'EPSG:4326',
+        extent,
+    });
+    source.zoom = { max: 20, min: 0 };
+
+    const layer = new Layer('foo', {
+        source,
+        projection: 'EPSG:4326',
+        info: { update: () => {} },
+        tileMatrixSets: [
+            'TMS:4326',
+            'TMS:3857',
+        ],
+        parent: { tileMatrixSets: [
+            'TMS:4326',
+            'TMS:3857',
+        ],
+        },
+    });
+
+    const nodeLayer = new MaterialLayer(material, layer);
+    nodeLayer.level = 10;
+
+    const node = new TileMesh(geom, material, layer, extent, 19);
+    node.layerUpdateState[layer.id] = new LayerUpdateState();
+    const state = node.layerUpdateState[layer.id];
+    state.lowestLevelError = layer.source.zoom.max;
+    const failureParams = state.failureParams;
+    const history = [];
+    let LOADED = false;
+    let maxLevel = 13;
+
+    function isExist(level, currentLevel) {
```

Comment:
Could you rename this method ? To understand easily the test you are doing ? Thanks

Reply:
renamed